unumpy/_multimethods.py

```diff
@@ -453,6 +453,48 @@ def asanyarray(a, dtype=None, order=None):
     return (a, mark_dtype(dtype))
 
 
+def _asfarray_default(a, dtype=float):
+    a = asarray(a, dtype=dtype)
+    if a.dtype.name not in ("float16", "float32", "float64"):
```

Comment:
```suggestion
    if a.dtype.name.startswith("float"):
```

Reply:
This should be:

```python
if not a.dtype.name.startswith("float"):
    dtype = float
```

If I'm not mistaken.